### PR TITLE
BF: make Simple_Prep_docker run on OSX

### LINKED ISSUE #17: Simple_Prep_docker not running on MacOS 10.12.3 (Sierra) ?
Trying Simple_Prep_docker jessie on Dave new mac  : seems that readlink, sed ...etc have different options. Bash looks lie 3.2.57 while my machine would 4.3.46(1)-release (x86_64-pc-linux-gnu)
Installing bash 4.?? and changing the shebang doesn't seem to help though...

### PR DESCRIPTION
**Pull request by yarikoptic**
as @khinsen  pointed out -- dockerfile preparation script doesn't work on OSX .  This is a PR to address that

- [x]  no sed -i and readlink -f
- [x] check if it works and fix the rest of OSX compatibility issues
- [x] double check back on Linux systems that it works as expected

Closes #17 
Relates to #15

### CODE REVIEW

**Comment by yarikoptic** [comment]
@satra -- I have no working docker on OSX, so could just verify that it seems to generate correct Dockerfile and then verify that it works on Linux.  Could you give it a shot on OSX following current(modified) README.md

**Comment by satra** [comment]
will do - also added list of things to take care of today - #20.

**Comment by satra** [comment]
on osx: 

```
$ ./Simple_Prep_docker Simple_Prep_docker-Dockerfile.in 
readlink: illegal option -- f
usage: readlink [-n] [file ...]
```

let me first update docker file before you make anymore changes.

**Comment by yarikoptic** [comment]
Are you sure you are in my branch? Check on invocation in readme - needs release  and a date

**Comment by satra** [comment]
ah no. i was blindly using master.

File: Simple_Prep_docker [hunk 1]
```diff
@@ -34,18 +44,25 @@ sed -e "s,DL_DIST,$DL_DIST,g" \
 
 if [ ! -z "$DL_DATE" ]; then
 {
-    # replace with snapshots
-    echo "sed -i -e 's,http://neuro.debian.net/debian/* ,http://snapshot-neuro.debian.net:5002/archive/neurodebian/$DL_DATE/ ,g' etc/apt/sources.list.d/neurodebian.sources.list;"
+    # replace with snapshots.  Cannot use -i due to OSX but since done inside the container
+    # should be safe just to use some explicit /tmp/apt.sources
+    echo "sed -e 's,http://neuro.debian.net/debian/* ,http://snapshot-neuro.debian.net:5002/archive/neurodebian/$DL_DATE/ ,g' etc/apt/sources.list.d/neurodebian.sources.list > /tmp/apt.sources; mv /tmp/apt.sources etc/apt/sources.list.d/neurodebian.sources.list"
```
Comment (yarikoptic) [inline Simple_Prep_docker]: Actually this change is not needed... I was blinded by the osx pain :-)

**Review by yarikoptic — COMMENTED** [review]
Needs a minor revert for cleanness

**Comment by yarikoptic** [comment]
pushed that little revert... @satra -- could you check if all is kosher on OSX so it could finally be merged/forgotten about? ;)

**Comment by satra** [comment]
works on my OSX version: 10.12.4 (16E195)

**Pull request merged**